pathio.go

```diff
@@ -92,6 +98,61 @@ func (c *Client) WriteReader(path string, input io.ReadSeeker) error {
 	return writeToLocalFile(path, input)
 }
 
+// ListFiles lists all the files/directories in the directory. It does not recurse
+func (c *Client) ListFiles(path string) ([]string, error) {
+	if strings.HasPrefix(path, "s3://") {
+		s3Conn, err := s3ConnectionInformation(path)
+		if err != nil {
+			return nil, err
+		}
+		return lsS3(s3Conn)
+	}
+	return lsLocal(path)
+}
+
+func lsS3(s3Conn s3Connection) ([]string, error) {
+	params := s3.ListObjectsInput{
+		Bucket:    aws.String(s3Conn.bucket),
+		Prefix:    aws.String(s3Conn.key),
+		Delimiter: aws.String("/"),
+	}
+	finalResults := []string{}
+
+	for {
+		resp, err := s3Conn.handler.ListObjects(&params)
+		if err != nil {
+			return nil, err
+		}
+		results := make([]string, len(resp.Contents)+len(resp.CommonPrefixes))
+		for i, val := range resp.CommonPrefixes {
+			results[i] = *val.Prefix
+		}
+		for i, val := range resp.Contents {
+			results[i+len(resp.CommonPrefixes)] = *val.Key
+		}
+		finalResults = append(finalResults, results...)
+
+		if *resp.IsTruncated {
```

Comment:
When we next go through the for loop, the changed params will be used on line 122.

I should probably add a comment explaining why we do this.